pkg/util/util.go

```diff
@@ -17,22 +17,25 @@ limitations under the License.
 package util
 
 import (
-	gobuild "go/build"
+	"fmt"
+	"golang.org/x/mod/modfile"
+	"io/ioutil"
+	"os"
 	"path/filepath"
 	"strings"
 )
 
 // CurrentPackage returns the go package of the current directory, or "" if it cannot
 // be derived from the GOPATH.
 // This logic is taken from k8.io/code-generator, but has a change of letting user pass the
-// directory whose pacakge is to be found.
+// directory whose package is to be found.
 func CurrentPackage(dir string) string {
-	for _, root := range gobuild.Default.SrcDirs() {
-		if pkg, ok := hasSubdir(root, dir); ok {
-			return pkg
-		}
+	gomod, err := ioutil.ReadFile(filepath.Join(dir, "go.mod"))
```

Comment:
This errors out when I try running this inside the repository which has apis. Also this requires that the generator is run from $GOPATH always. Is that intended? With code-generator it makes us specify [outputDir](https://github.com/kubernetes/code-generator/blob/65c70a51ae8eca63d4b9ab1e80fe636097a6ce63/cmd/applyconfiguration-gen/args/args.go#L59-L61) with respect to $GOPATH, and that makes it running outside the repository easier. Can we follow that approach?

Reply:
I'll try that if that works and dig into what else code-generator does for this case and update this PR

Reply:
I don't have my code in gopath but I didn't try running from within the repo. We'll want to support both